tests/zfs-tests/tests/functional/mmp/mmp_hostid.ksh

```diff
@@ -43,6 +43,9 @@ function cleanup
 	log_must rm $MMP_DIR/file.{0,1,2,3,4,5}
 	log_must rmdir $MMP_DIR
 	log_must mmp_clear_hostid
```

Comment:
Sorry, I'm still unclear why this is needed.  This line calls `mmp_clear_hostid` unconditionally which should do exactly what the block you've added does.

```ksh
function mmp_clear_hostid
{
        rm -f $HOSTID_FILE
}
```

The only way I can see that this line won't run is if one of the previous `log_must` lines fails.  Given that, what I was suggested was something like this:

```diff
diff --git a/tests/zfs-tests/tests/functional/mmp/mmp_hostid.ksh b/tests/zfs-test
index b492b10..bb13850 100755
--- a/tests/zfs-tests/tests/functional/mmp/mmp_hostid.ksh
+++ b/tests/zfs-tests/tests/functional/mmp/mmp_hostid.ksh
@@ -40,9 +40,9 @@ verify_runnable "both"
 function cleanup
 {
        default_cleanup_noexit
-       log_must rm $MMP_DIR/file.{0,1,2,3,4,5}
-       log_must rmdir $MMP_DIR
-       log_must mmp_clear_hostid
+       rm $MMP_DIR/file.{0,1,2,3,4,5}
+       rmdir $MMP_DIR
+       mmp_clear_hostid
 }
 
 log_assert "Verify hostid file can reside on a ZFS dataset"
```

Reply:
about /etc/hostid file. on DilOS we have service system/hostid where we create /etc/hostid file if it missed with boot/reboot. we have this file all the time. i try prepare some universal way for linux and dilos, where if test failed, we have symlink to file /etc/hostid present and next ZTS iteration in loop will be failed. i try to remove link to file /etc/hostid only in this test, where we create it for reduce issues with others tests what can be done before clean.sh step